src/biguint.rs

```diff
@@ -143,6 +144,72 @@ impl fmt::Octal for BigUint {
     }
 }
 
+impl fmt::LowerExp for BigUint {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Option::Some(small) = self.to_u64() {
+            fmt::LowerExp::fmt(&small, f)
+        } else {
+            // The max digits that can fit into a f64, which ensures
+            // that printing won't have rounding errors.
+            const MAX_FLOAT_DIGITS: usize = 14;
+            if f.precision().filter(|x| x < &MAX_FLOAT_DIGITS).is_some() {
+                let as_f64 = self.to_f64().unwrap();
+                if as_f64.is_finite() {
+                    fmt::LowerExp::fmt(&as_f64, f)
+                } else {
+                    fmt_exp_slow(self, f, false, true)
+                }
+            } else {
+                fmt_exp_slow(self, f, false, true)
+            }
+        }
+    }
+}
+
+impl fmt::UpperExp for BigUint {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let Option::Some(small) = self.to_u64() {
+            fmt::UpperExp::fmt(&small, f)
+        } else {
+            // The max digits that can fit into a f64, which ensures
+            // that printing won't have rounding errors.
+            const MAX_FLOAT_DIGITS: usize = 14;
```

Comment:
`f64::DIGITS` is 15, though it does say "approximate". Perhaps we should us that, minus one?

Reply:
I subtracted 1 to ensure the rounding wouldn't turn out weird, but forgot to update the docs. I didn't know `f64::DIGITS` existed, so I'll update to take that into account.